README.md

```diff
@@ -1,11 +1,87 @@
 # Topiary
 
-![CI Status](https://github.com/tweag/topiary/actions/workflows/ci.yml/badge.svg)
+![CI Status][ci-badge]
 
 Topiary aims to be a uniform formatter for simple languages, as part of
-the [Tree-sitter](https://tree-sitter.github.io/tree-sitter/) ecosystem.
-It is named after the art of clipping or trimming trees into fantastic
-shapes.
+the [Tree-sitter] ecosystem. It is named after the art of clipping or
+trimming trees into fantastic shapes.
+
+## Motivation
+
+The style in which code is written has, historically, been mostly left
+to personal choice. Of course, this is subjective by definition and has
+led to many wasted hours reviewing formatting choices, rather than the
+code itself. Prescribed style guides were an early solution to this,
+spawning tools that lint a developer's formatting and ultimately leading
+to automatic formatters. The latter were popularised by
+[`gofmt`][gofmt], whose developers had the insight that "good enough"
+uniform formatting, imposed on a codebase, with no scope for subjective
+adjustment, largely resolves these problems.
+
+Topiary follows this trend by aspiring to be a "universal formatter
+engine", which allows developers to not only automatically format their
+codebases with a uniform style, but to define that style for new
+languages using a [simple DSL][tree-sitter-query]. This allows for the
+rapid prototyping of formatters, providing a [Tree-sitter
+grammar][tree-sitter-parsers] is defined for that language.
+
+## Design Principles
+
+Topiary has been created with the following goals in mind:
+
+* To target languages for which a [Tree-sitter
+  grammar][tree-sitter-parsers] exist; whether officially supported by
+  the Tree-sitter project, or otherwise.
+
+* To support a handful of languages, with "one true formatting style"
+  for each, where no de facto formatter exists. These supported
+  formatting styles must:
+
+  * Be compatible with attested formatting styles used for that language
+    in the wild.
+
+  * Be faithful to the author's intent, in terms of line spacing.
+
+  * Result in minimal diffs.
+
+  * Be idempotent. That is, formatting of already-formatted code doesn't
+    change anything.
+
+* Code must be written in such a way that it's easy to modify and
+  maintain.
+
+* Code and formatting styles must be well-tested and robust, so that the
+  formatter can be used in large projects.
+
+* For end users -- i.e., not formatting style developers -- the
+  formatter should:
+
+  * Not permit any subjective configuration.
```

Comment:
Again :slightly_smiling_face:

Reply:
IMO, the lack of configurability -- as an end user -- is important; its necessary for delivering on uniformity. Obviously the styles are customisable if you're so inclined, but I think that doing so pushes you into a different user category.

Reply:
Reworded in fd18ebc1c1a